stdlib/src/memory/reference.mojo

```diff
@@ -332,3 +332,31 @@ struct Reference[
             The MLIR reference for the Mojo compiler to use.
         """
         return __get_litref_as_mvalue(self.value)
+
+    @always_inline("nodebug")
+    fn __eq__(self, rhs: Reference[type, _, address_space]) -> Bool:
+        """Returns True if the two pointers are equal.
+
+        Args:
+            rhs: The value of the other pointer.
+
+        Returns:
+            True if the two pointers are equal and False otherwise.
+        """
+        return UnsafePointer(
+            __mlir_op.`lit.ref.to_pointer`(self.value)
+        ) == __mlir_op.`lit.ref.to_pointer`(rhs.value)
```

Comment:
Maybe wrap both with `UnsafePointer` instead of relying on implicit conversion.

Reply:
it looks so nice though

Reply:
I'll consider it. There might be a better way anyways, if not soon.